drivers/sensor/realtek/realtek_tach_rts5912/tach_realtek_rts5912.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2023 Realtek, SIBG-SD7
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_tach
+
+#include <errno.h>
+#include <soc.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_rts5912.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/logging/log.h>
+
+#include "reg/reg_tacho.h"
+
+LOG_MODULE_REGISTER(tach_rts5912, CONFIG_SENSOR_LOG_LEVEL);
+
+struct tach_rts5912_config {
+	TACHO_Type *regs;
+	uint32_t clk_grp;
+	uint32_t clk_idx;
+	uint32_t clk_src;
+	uint32_t clk_div;
+	const struct device *clk_dev;
+	const struct pinctrl_dev_config *pcfg;
+	int pulses_per_round;
+};
+
+struct tach_rts5912_data {
+	uint16_t count;
+};
+
+#define COUNT_100KHZ_SEC  100000U
+#define SEC_TO_MINUTE     60U
+#define PIN_STUCK_TIMEOUT 100U
+
+int tach_rts5912_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	ARG_UNUSED(chan);
+
+	const struct tach_rts5912_config *const cfg = dev->config;
+	struct tach_rts5912_data *const data = dev->data;
+	TACHO_Type *const tach = cfg->regs;
+
+	uint8_t poll_count = 0;
+
+	tach->STS = TACHO_STS_CNTRDY_Msk;
+
+	while (poll_count < PIN_STUCK_TIMEOUT) {
```

Comment:
Consider using `WAIT_FOR`, something like (I did not check if this actually work):
```suggestion
	if(WAIT_FOR(tach->STS & TACHO_STS_CNTRDY_Msk, PIN_STUCK_TIMEOUT * USEC_PER_MSEC, k_msleep(1)) {
		tach->STS = TACHO_STS_CNTRDY_Msk;
		data->count = (tach->CTRL & TACHO_CTRL_CNT_Msk) >> TACHO_CTRL_CNT_Pos;
	} else {
		data->count = 0;
	}
```

Reply:
done, that work fine, thank you.